internal/v5/api.go

```diff
@@ -114,10 +125,13 @@ func New(pool *charmstore.Pool, config charmstore.ServerParams, rootPath string)
 	var permChecker permChecker
 	if config.IdentityAPIURL != "" && config.AgentUsername != "" && config.AgentKey != nil {
 		logger.Infof("group membership enabled at %s; agent %s", config.IdentityAPIURL, config.AgentUsername)
-		identityClient := idmclient.New(idmclient.NewParams{
+		identityClient, err := idmclient.New(idmclient.NewParams{
 			BaseURL: config.IdentityAPIURL,
 			Client:  agent.NewClient(config.AgentUsername, config.AgentKey),
 		})
+		if err != nil {
+			panic(err)
```

Comment:
Strictly speaking I'd say that as the IdentityAPIURL originally came from user input, the error case here cannot be attributed to being a bug and so a panic isn't appropriate. I suspect we should rethink the idmclient API if we need this to complete without errors.

On the other hand I know how hard it is to fail to parse a URL, so I'm not that worried about it.